File: cogs/invite_leaderboard.py
import discord
from discord.ext import commands
from discord import app_commands

class InviteLeaderboard(commands.Cog):
    invites_group = app_commands.Group(
        name="invites",
        description="Invite management commands"
    )

    def __init__(self, bot):
        self.bot = bot

    def cog_unload(self):
        pass

    @invites_group.command(name="leaderboard", description="Show the top invites leaderboard.")
    async def invite_leaderboard(self, interaction: discord.Interaction):
        await interaction.response.defer(ephemeral=False)

        invites = await interaction.guild.invites()
        if not invites:
            await interaction.followup.send("No invites found for this server.")
            return

        # Sort invites by uses descending
        sorted_invites = sorted(invites, key=lambda i: i.uses or 0, reverse=True)

        leaderboard_lines = []
        for idx, invite in enumerate(sorted_invites[:10], start=1):
            inviter = invite.inviter.name if invite.inviter else "Unknown"
            uses = invite.uses or 0
            leaderboard_lines.append(f"**{idx}.** {inviter} - `{uses}` uses (Code: `{invite.code}`)")

        leaderboard_text = "\n".join(leaderboard_lines)
        embed = discord.Embed(
            title="📊 Invite Leaderboard",
            description=leaderboard_text,
            color=discord.Color.blue()
        )
        await interaction.followup.send(embed=embed)

async def setup(bot):
    await bot.add_cog(InviteLeaderboard(bot))

File: commands/inviteLeaderboard.js
const { 
    SlashCommandBuilder, 
    EmbedBuilder, 
    Colors 
} = require('discord.js');

module.exports = {
    data: new SlashCommandBuilder()
        .setName('invites')
        .setDescription('Invite management commands')
        .addSubcommand(subcommand =>
            subcommand
                .setName('leaderboard')
                .setDescription('Show the top invites leaderboard.')
        ),

    async execute(interaction) {
        const subcommand = interaction.options.getSubcommand();
        
        if (subcommand === 'leaderboard') {
            await this.showLeaderboard(interaction);
        }
    },

    async showLeaderboard(interaction) {
        await interaction.deferReply({ ephemeral: false });
        
        try {
            // Fetch all guild invites
            const invites = await interaction.guild.invites.fetch();
            
            if (!invites || invites.size === 0) {
                return await interaction.followUp("No invites found for this server.");
            }
            
            // Convert to array and sort by uses descending
            const sortedInvites = [...invites.values()]
                .sort((a, b) => (b.uses || 0) - (a.uses || 0))
                .slice(0, 10); // Get top 10
            
            const leaderboardLines = sortedInvites.map((invite, index) => {
                const inviterName = invite.inviter ? invite.inviter.username : 'Unknown';
                const uses = invite.uses || 0;
                return `**${index + 1}.** ${inviterName} - \`${uses}\` uses (Code: \`${invite.code}\`)`;
            });
            
            const embed = new EmbedBuilder()
                .setTitle('📊 Invite Leaderboard')
                .setDescription(leaderboardLines.join('\n'))
                .setColor(Colors.Blue)
                .setFooter({ text: `${interaction.guild.name} | Total invites: ${invites.size}` })
                .setTimestamp();
            
            await interaction.followUp({ embeds: [embed] });
        } catch (error) {
            console.error('Error fetching invites:', error);
            await interaction.followUp('Failed to fetch invite information. Make sure I have the "Manage Server" permission.');
        }
    }
};